rust/src/array.rs

```diff
@@ -0,0 +1,248 @@
+    }
+}
+
+impl From<Vec<f64>> for Array {
+    fn from(v: Vec<f64>) -> Self {
+        Array { len: v.len() as i32, null_count: 0, null_bitmap: Bitmap::new(v.len()), data: ArrayData::Float64(v) }
+    }
+}
+
+impl From<Vec<i32>> for Array {
+    fn from(v: Vec<i32>) -> Self {
+        Array { len: v.len() as i32, null_count: 0, null_bitmap: Bitmap::new(v.len()), data: ArrayData::Int32(v) }
+    }
+}
+
+impl From<Vec<Option<i32>>> for Array {
+    fn from(v: Vec<Option<i32>>) -> Self {
+        let mut null_count = 0;
+        let mut null_bitmap = Bitmap::new(v.len());
+        for i in 0 .. v.len() {
+            if v[i].is_none() {
+                null_count+=1;
+                null_bitmap.set(i);
+            }
+        }
+        let values = v.iter().map(|x| x.unwrap_or(0)).collect::<Vec<i32>>();
+        Array { len: v.len() as i32, null_count, null_bitmap, data: ArrayData::Int32(values) }
+    }
+}
+
+impl From<Vec<i64>> for Array {
+    fn from(v: Vec<i64>) -> Self {
+        Array { len: v.len() as i32, null_count: 0, null_bitmap: Bitmap::new(v.len()), data: ArrayData::Int64(v) }
+    }
+}
+
+impl From<Vec<Option<i64>>> for Array {
+    fn from(v: Vec<Option<i64>>) -> Self {
+        let mut null_count = 0;
+        let mut null_bitmap = Bitmap::new(v.len());
+        for i in 0 .. v.len() {
+            if v[i].is_none() {
+                null_count+=1;
+                null_bitmap.set(i);
+            }
+        }
+        let values = v.iter().map(|x| x.unwrap_or(0)).collect::<Vec<i64>>();
+        Array { len: v.len() as i32, null_count, null_bitmap, data: ArrayData::Int64(values) }
+    }
+}
+
+/// This method mostly just used for unit tests
+impl From<Vec<&'static str>> for Array {
+    fn from(v: Vec<&'static str>) -> Self {
+        Array::from(v.iter().map(|s| s.to_string()).collect::<Vec<String>>())
+    }
+}
+
+impl From<Vec<String>> for Array {
+    fn from(v: Vec<String>) -> Self {
+        let mut offsets : Vec<i32> = Vec::with_capacity(v.len() + 1);
+        let mut buf = BytesMut::with_capacity(v.len() * 32);
+        offsets.push(0_i32);
+        v.iter().for_each(|s| {
+            buf.put(s.as_bytes());
+            offsets.push(buf.len() as i32);
+        });
+        Array {
+            len: v.len() as i32,
+            null_count: 0,
+            null_bitmap: Bitmap::new(v.len()),
+            data: ArrayData::Utf8(ListData { offsets, bytes: buf.freeze() })
+        }
+    }
+}
+
+impl From<Vec<Rc<Array>>> for Array {
+    fn from(v: Vec<Rc<Array>>) -> Self {
+        Array {
+            len: v.len() as i32,
+            null_count: 0,
+            null_bitmap: Bitmap::new(v.len()),
+            data: ArrayData::Struct(v.iter().map(|a| a.clone()).collect())
+        }
+    }
+}
+
+/// List of variable-width data such as Utf8 strings
+pub struct ListData {
+    pub offsets: Vec<i32>,
+    pub bytes: Bytes
+}
```

Comment:
How about placing it in the `ArrayData` enum, something like [here](https://github.com/kszucs/arrow/blob/enum/rust/src/e.rs#L66)?